### PR TITLE
Data Types, Operators and Expressions

### LINKED ISSUE #13: Data Types, Operators and Expressions
N | 问题描述 | 输出格式
----|------|------
1 | 交换2个数x和y，用3+种不同方法来实现 | `{x:d} {y:d} => {:d} {:d}`
2 | 判断某个年份year是否为润年 | `{year:d} => {:d}` 
3 | 判断2个浮点数a和b是否相等 | `{a:f} {b:f} {dt:f} => {:d}`
4 | 判断一个整数n是否为偶数 (even) | `{n:d} => {:d}` 
5 | n为一整数，不用乘除及取模求nx17、n%8、n/16 | `{n:d} => {:d} {:d} {:d}`
6 | 判断一个整数n是否为2<sup>k</sup>(2的k次方) | `{n:d} => {:d} `
7 | 位运算分别实现`chmod 600/+x/u+w/go-w` 4种操作 | `{m:o} => {:o} {:o} {:o} {:o}`

**相应说明**:
1. 同时在PR页面描述各方法的优缺点
1. 输出部为"1/0"，表示“True/False”
1. 输出部为"1/0"，表示“相等/不等”
1. 输出部为"1/0"，表示"even/odd"。要求分别用算术运算符和位运算符实现
1. (无)
1. 输出部为"1/0"，表示“True/False”。n要求覆盖“大于、小于、等于”0三种情况
1. 输出部4个new m均基于m初始值计算而得

Fork #4

### PR DESCRIPTION
**Pull request by dark-w**
Resolve #13

Signed-off-by: dark-w <dkfire30@outlook.com>

### CODE REVIEW

File: operators/dark-w.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#Data Types, Operators and Expressions
+
+a = 11
+b = 22
+
+#优：速度快 缺点：多分配了内存
+print('{0:d} {1:d}'.format(a, b))
```
Comment (conke) [inline operators/dark-w.py]: 简单实用{}即可，无需其他参数（下同）

File: operators/dark-w.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+n2 = 10
+ans1 = n2 & 7
+print(n2, ans1)
+
+n3 = 0b11100000
+ans2 = n3 >> 4
+print(n3, ans2)
+
+m = 0o644
+m1 = m ^ m ^ 0o600
```
Comment (bonnie-young) [inline operators/dark-w.py]: 这里想复杂了，直接复制即可，m1 = 0o600

**Comment by dark-w** [comment]
No. | Output | Format | Result
--- | ------ | ------ | ------
1 | 11 22 => 22 11 | {} {} => {} {} | True
2 | 22 11 => 11 22 | {} {} => {} {} | True
3 | 11 22 => 22 11 | {} {} => {} {} | True
4 | 22 11 => 11 22 | {} {} => {} {} | True
5 | 1.6667 1.6668 0.001 eq | {} {} {} {} | True
6 | 2016 yes | {} {} | True
7 | 5 odd | {} {} | True
8 | 5 odd | {} {} | True
9 | 8 yes | {} {} | True
10 | 3 x 17 = 51 | {} x 17 = {} | True
11 | 10 % 8 = 2 | {} % 8 = {} | True
12 | 224 / 16 = 14 | {} / 16 = {} | True
13 | 644: 600 755 644 644 | {}: {} {} {} {} | True

File: operators/dark-w.py (hunk 2 above)
Comment (conke) [reply operators/dark-w.py]: 请按博妮方式更正

**Comment by dark-w** [comment]
No. | Output | Format | Result
--- | ------ | ------ | ------
1 | 11 22 => 22 11 | {} {} => {} {} | True
2 | 22 11 => 11 22 | {} {} => {} {} | True
3 | 11 22 => 22 11 | {} {} => {} {} | True
4 | 22 11 => 11 22 | {} {} => {} {} | True
5 | 1.6667 1.6668 0.001 eq | {} {} {} {} | True
6 | 2016 yes | {} {} | True
7 | 5 odd | {} {} | True
8 | 5 odd | {} {} | True
9 | 8 yes | {} {} | True
10 | 3 x 17 = 51 | {} x 17 = {} | True
11 | 10 % 8 = 2 | {} % 8 = {} | True
12 | 224 / 16 = 14 | {} / 16 = {} | True
13 | 644: 600 755 644 644 | {}: {} {} {} {} | True

**Pull request merged**